src/components/PlayerProfile.vue

```diff
@@ -146,17 +137,11 @@ export default {
       return this.formatDate(this.player.lastLogin * 1);
     },
     playerRanks() {
-      if (!this.player?.groups?.length) return 'None';
-      return `[${this.player.groups.map((group) => group.id)
-        .join()
-        .split(',')
-        .join('] [')
-        .toUpperCase()}]`;
+      if (!this.player?.groups?.length) return ['None'];
+      return this.player.groups.map((group) => group.id.toUpperCase());
     },
     playerHistoryData() {
-      if (this.player === null || this.player === undefined) return null;
-      if (this.player.infractions === undefined) return null;
-      if (this.player.infractions.bans.length === 0) return null;
+      if (!this.player?.infractions?.bans?.length) return null;
```

Comment:
You're null checking `this.player` in multiple places. For the player == null case,  would it make sense to use a `v-if="this.player` in the markup so we don't even call these methods in the first place?